day1/day_1.erl

```diff
@@ -0,0 +1,36 @@
+-module(day_01).
+-export([calc_fuel_for_modules/0, calc_total_fuel/0]).
+
+read_input(FileName) ->
+    {ok, Data} = file:read_file(FileName),
+    lists:delete(<<>>, binary:split(Data, [<<"\n">>], [global])).
+
+process_input(L) ->
+    [ list_to_integer(binary_to_list(X)) || X <- L ].
+
+calc_module_fuel(X) ->
+    X div 3 - 2.
+
+calc_fuel_for_fuel(X) ->
+    calc_fuel_for_fuel(X, 0).
+
+calc_fuel_for_fuel(X, Total) ->
+    Fuel = X div 3 - 2,
```

Comment:
In Erlang, we love pattern-matching directly rather than these indirect boolean checks. So I might rewrite this as:

```erlang
calc_fuel_for_fuel(X, Total) ->
    case calc_module_fuel(X) of
        Additional when Additional =< 0 -> Total;
        Additional -> calc_fuel_for_fuel(Additional, Total + Additional)
    end.
```

Because you may have to come back to this to support more than 2 branches. Also I think `Fuel >= 0` will match 0 for Fuel and make an extra function call (negligible but noticed).

Reply:
Updated my solution to utilize pattern matching when possible.